cmake/ExternalDependencies.cmake

```diff
@@ -73,3 +73,23 @@ endif()
 
 # Make all declared dependencies available.
 FetchContent_MakeAvailable(${FETCH_PACKAGES})
+
+if(ENABLE_MQT_CORE_MLIR_CATALYST_PLUGIN)
+  # Manually detect the installed Catalyst Python and get its cmake directory.
+  execute_process(
+    COMMAND "${Python_EXECUTABLE}" -c
+            "import catalyst, os; print(os.path.dirname(catalyst.__file__))"
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    OUTPUT_VARIABLE Python_Catalyst_DIR
+    ERROR_QUIET)
+  message(STATUS "Found Catalyst package: ${Python_Catalyst_DIR}")
+
+  # TODO: once the Catalyst Python package provides the necessary files set(Catalyst_DIR
+  # "${Python_Catalyst_DIR}/mlir/build/lib/cmake/catalyst")
```

Comment:
Ideally, there would be no need to do something like `set(Catalyst_DIR "${Python_Catalyst_DIR}/...`, but `Catalyst_DIR` would already be a result of the `execute_process` call.
Either catalyst provides an entry point for that (similar to pybind, nanobind, or mqt-core), or we parse this somewhat from output similar to the existing command here by a simple Python script.